### DISCUSSION TITLE
提了几个关于“稍后再看”和“个性推荐”的想法

### DISCUSSION THREAD

**Discussion by nonworry** [BewlyBewly/BewlyBewly#432] [Ideas]
刚刚下载来使用，首先感谢下开发者的这个插件，做得很好看，功能也挺齐全的，而且加载速度跟我没用插件时的差不多（虽然因为电脑性能原因吧，本来加载就有点慢的，不过用了插件没变得更卡那就很好了）。然后因为原本我用的是另一个油猴上的首页插件bilibili-app-recommend，所以对比了下，感觉“稍后再看”这边或许还有改进空间，然后我顺便想再提个那个插件里没有的功能，也都罗列在下面了。
1、目前这个插件里的稍后再看感觉如果能跟首页里的视频卡片一样一行5个就更好了，因为有点喜欢囤积视频，通常是把喜欢的视频放稍后再看，之后再统一在空闲时间点开看，而目前这个是一列下来的，感觉要找视频的话，相对来说效率较低，或许可以考虑把右侧三个功能键放到上面“稍后再看”的标题文字边上？
2、再一个是bilibili-app-recommend插件里的一个功能，可以把稍后再看里选中的视频“重新加入稍后再看”，就是相当于把视频放到第一位，这个我使用得最多了，因为习惯洗碗或洗澡时看视频，所以通常会在这之前，先通过电脑把稍后再看里待会想看的视频重新排列下，然后拿起手机直接点开来看，之后看完一个视频就不用再去选择下一个视频想看什么了（毕竟有时候稍后再看里的视频会是些文本量较多或需要看画面的时候较多的视频，而做事情的时候难免会遗漏，所以我通常会需要挑选些比较轻松的视频排列好顺序方便自动播放）
3、稍后在看里，如果有个按视频时长排序的功能就更好了。因为自从稍后在看解开100个视频的限制后，囤积太多了，一直想看看清一清，而如果能有个时长排列，也方便先把时长短的看掉。
4、感觉这个或许比较麻烦，而且也不知道其他用户用得多不多，如果不行的话也没关系。还是关于“重新添加稍后在看”这个功能，就是在稍后在看页面里，多加一个选择按钮，点击后下面的视频卡片进入待选择状态，然后选择几个视频，就在功能按钮边上实时显示出已选中的视频的总时长，选择完毕后再点击按钮确定选择，那刚刚选好的视频就按选择时的顺序，按序重新添加进稍后在看。总时长是因为像我做一些比较固定时长的事情时想边看视频边做，可能是半小时或是一小时，那我通常会是一边给稍后再看的视频做排序，一边计算这几个视频加起来的总时长够不够我待会的时长，不然添加少了，做事情做到一半还得找视频看。而“按选择时的顺序，按序重新添加进稍后在看”则是因为一个一个添加的时候，其实在待会播放时是相当于倒着添加时的顺序在播放的，这个感觉就有点说不上来的不舒服，如果要比喻的话，就像是酒店反着上菜，先给你上水果，再上正餐，到最后再给人上开胃菜哈哈。
5、个性推荐里，右下角有个回到最上面的按钮，同时回到最上面后会变成刷新按钮，这个按钮如果鼠标放在按钮上，滚轮会用不了，虽然是个小问题，不过发现了就顺便提一下。
6、“个性推荐”的这个选项卡，不知道能不能做个单击和双击功能？例如单击还是现在这样打开“个性推荐”视频列表，而双击则是在打开“个性推荐”的同时，触发刷新功能。毕竟可能我在个性推荐里往下刷了好一会了，点了下“正在关注”看了下关注的up主的视频，然后又点回“个性推荐”，就已经回到最上面的位置了，这时候要刷新推荐，又得把鼠标移到右下角，感觉有点远。
7、首页上的选项卡不知道能不能多开一个直播页，可以用来查看目前正在直播的关注的主播。

**Comment by hakadao** [comment]
十分感謝支持了，很少有人那麼認真寫出如此詳細的建議與想法，感謝您珍貴的建議！！！！
1. 三個按鈕切換排列佈局其實有點麻煩，目前稍後再看和首頁的卡片是用兩種實現方法，重構不是太方便
2. 這個功能待考慮
3. 時長排序功能待考慮，這部分要放在用戶瀏覽器處理我不知道有些機型能否可以承受得了
4. 如上待考慮功能，另外這種操作想法恪守很有些，結合實際考慮能否實現
5. 感覺這個好像也沒有特別嚴重問題，暫時不管了，但是感謝閣下細心觀察
6. 這個不打算，目前主流網站的操作邏輯也是這麼處理，這裡單獨處理的話不是所有人都一定接受到。但是現在個性推介你就算是切換到上面其他按鈕後再回來數據會保留我做了處理
7. 其實我不怎麼看直播也不清楚如何做好直播頁

最後非常感謝閣下的寶貴建議和想法，十分有誠意的寫下自己的一些建議，不過目前要做的任務和修正的錯誤以及著重的後續新功能有點多，後面有時間我會細心研究，最後非常感謝閣下的提議🙏🙏🙏
